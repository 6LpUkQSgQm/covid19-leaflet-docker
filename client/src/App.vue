<template>
<div>
  <v-app style="background-color:#011627 !important">
    <Navigation />
    <Dashboard />
    <Footer />
  </v-app>
  </div>
</template>

<script>
import Navigation from "@/layouts/Navigation";
import Footer from "@/layouts/Footer";
import Dashboard from "@/pages/Dashboard";

export default {
  components: {
    Navigation,
    Dashboard,
    Footer
  },
};
</script>
