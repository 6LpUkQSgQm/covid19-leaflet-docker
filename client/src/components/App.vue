<template>
  <v-app>
    <Navigation />
    <Map v-if="markersCountries.length" />
    <div style="background-color:grey;height:91%" v-else>
      <div style="position:absolute;margin-top:20%; margin-left:48%;">
        <v-progress-circular :size="50" color="secondary" indeterminate />
        <p style="font-size:12px;">Loading...</p>
      </div>
    </div>
    <Footer />
  </v-app>
</template>

<script>
import Navigation from "@/components/layouts/Navigation";
import Footer from "@/components/layouts/Footer";
import Map from "@/components/pages/Map";
import { mapGetters } from "vuex";

export default {
  components: {
    Navigation,
    Map,
    Footer,
  },
  computed: {
    ...mapGetters(["markersCountries"]),
  },
  created() {
    this.$store.dispatch("getDataMarkersCountries");
  },
  mounted() {
    this.$ga.enable();
  }
};
</script>
