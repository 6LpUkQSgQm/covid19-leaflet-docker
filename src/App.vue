<template id="app">
  <v-app
    class="my-app"
    :style="{ background: $vuetify.theme.themes[theme].background }"
  >
    <Navigation />
    <Dashboard />
    <Footer />
  </v-app>
</template>

<script>
import Navigation from "@/layouts/Navigation";
import Footer from "@/layouts/Footer";
import Dashboard from "@/pages/Dashboard";

export default {
  name: "App",
  components: {
    Navigation,
    Dashboard,
    Footer,
  },
  computed: {
    theme() {
      return this.$vuetify.theme.dark ? "dark" : "light";
    },
  },
};
</script>
<style>
.myapp {
  position: relative;
}
.my-app.v-application .primary--text {
  color: grey !important;
}
</style>
