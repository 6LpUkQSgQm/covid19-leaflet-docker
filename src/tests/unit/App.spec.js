import Vue from "vue";
import Vuetify from "vuetify";
import App from "../../App";
import { createLocalVue, shallowMount } from "@vue/test-utils";

describe("App.vue", () => {
  const localVue = createLocalVue();
  
  let vuetify;
  const vuetifyOptions = {};
  Vue.use(Vuetify);
  new Vue({
    vuetify: new Vuetify(vuetifyOptions),
  });
  beforeEach(() => {
    vuetify = new Vuetify();
  });

  it("should have a v-app", () => {
    const wrapper = shallowMount(App, {
      localVue,
      vuetify,
    });
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find(".v-app"));
  });
});
