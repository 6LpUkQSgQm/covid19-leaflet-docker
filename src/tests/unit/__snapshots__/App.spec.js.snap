// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue should have a v-app 1`] = `
<v-app-stub id="app" class="my-app">
  <navigation-stub></navigation-stub>
  <dashboard-stub></dashboard-stub>
  <footer-stub></footer-stub>
</v-app-stub>
`;
